dynamic_programming/minimum_tickets_cost.py

```diff
@@ -0,0 +1,61 @@
+"""
+Author  : Alexander Pantyukhin
+Date    : November 1, 2022
+
+Task:
+Given a list of days when you need to travel. Each day is int from 1 to 365.
+You are able to use tickets for 1 day, 7 days and 30 days.
+Each ticket has a cost.
+
+Find the minimum cost you need to travel every day in the given list of days.
+
+Implementation notes:
+implementation Dynamic Programming up bottom approach.
+
+The implementation was tested on the
+leetcode: https://leetcode.com/problems/minimum-cost-for-tickets/
+"""
+
+from typing import List
+
+"""
+Minimum Cost For Tickets
+Dynamic Programming: up -> down.
+"""
+
+
+def mincost_tickets(days: list[int], costs: list[int]) -> int:
+    """
+    >>> mincost_tickets([1,4,6,7,8,20], [2,7,15])
+    11
+    >>> mincost_tickets([1,2,3,4,5,6,7,8,9,10,30,31], [2,7,15])
+    17
+    >>> mincost_tickets([1,2,3,4,5,6,7,8,9,10,30,31], [2,90,150])
+    24
```

Comment:
Please add tests with zero, negative numbers, floating point numbers, and mixing empty lists and non-empty lists.

Reply:
added tests for different cases. The function expects integers values only no floats. Should we add anyway float tests anyway?

Reply:
Yes.  The caller can throw any garbage they want at your function.  Letters, dicts, sets, None so be ready.

Type hints are only hints and callers can do what they want.  This is called Negative User Testing.

Reply:
fixed